added troubleshooting to fix #275 (#282)
notaryproject/notaryproject.dev: content/en/docs/troubleshooting.md

Check the `Subject` info in the output. If it is the identity that signs the artifact, you need to add the `Subject` info into `trustedIdentities`.

> ## When I verify an artifact, I get the error 'signature verification failed for all the signatures associated with IMAGE'

Author: fixed

This may result from one of the following issues:

> - Not having a trust store configured or the trust store is not configured correctly.
> - Trust store is not readable or the certificates stored in the trust store are not readable
> - no certificates were stored in trust store
> 
> Rerun the verify command with `--verbose` to get more details. 

Reviewer: ```suggestion
```suggestion
This is expected error message when all the signatures associated with $IMAGE are not trusted.
There are some situation where misconfiguration of trust store might lead to above error message which includes:
- Trust store directory or the certificates stored in the trust store are not readable
- Trust store doesn't contains any certificate.
To debug this issues rerun the verify command with `--verbose` flag to get more details.
```

Author: fixed

> The following example shows the trust store `mystore` has not be configured correctly and can't be found:

Reviewer: ```suggestion
The following example shows the usecase where the trust store `mystore` doesn't exists:
```

Author: made a minor fix to clarify

```console

> $ notation verify $IMAGE -v

Reviewer: ```suggestion
$ notation verify $IMAGE --verbose
```

Reviewer 2: I think it's also acceptable to use a simple alias of the flag as an example.

Author: fixed

...
ERRO authenticity validation failed. Failure reason: error while loading the trust store, "/home/USER/.config/notation/truststore/x509/ca/mystore" does not exist
WARN Signature sha256:11223344556677889900aabbccddeeff11223344556677889900aabbccddeeff failed verification with error: error while loading the trust store, "/home/USER/.config/notation/truststore/x509/ca/mystore" does not exist
Error: signature verification failed for all the signatures associated with localhost:5001/net-monitor@sha256:000999888777666555444333222111fffeeedddcccbbbbaaa000999888777666
```

> The following example shows the certificate `mystore.crt` in the `mystore` trust store could not be accessed:

Reviewer: ```suggestion
The following example shows the usecase where notation couldn't read the certificate `mystore.crt` stored in `mystore` trust store:
```

Author: i think the current wording is fine as is

```console

> notation verify --v $IMAGE

Reviewer: ```suggestion
notation verify $IMAGE --verbose
```

Author: fixed

...
ERRO authenticity validation failed. Failure reason: error while loading the trust store, error while reading certificates from "/home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt": open "/home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt": permission denied
WARN Signature sha256:11223344556677889900aabbccddeeff11223344556677889900aabbccddeeff failed verification with error: error while loading the trust store, error while reading certificates from "/home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt": open /home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt: permission denied
Error: signature verification failed for all the signatures associated with localhost:5001/net-monitor@sha256:000999888777666555444333222111fffeeedddcccbbbbaaa000999888777666
```

> The following example shows there are no certificates in the `mystore` trust store.

Reviewer: ```suggestion
The following example shows the usecase where there is no certificate in the `mystore` trust store.
```

Author: made a minor wording change to clarify

```console

> $ notation verify $IMAGE -v

Reviewer: ```suggestion
$ notation verify $IMAGE --verbose
```

Author: fixed

```

The following example shows the certificate `mystore.crt` in the `mystore` trust store could not be accessed:

```console
notation verify --v $IMAGE
...
ERRO authenticity validation failed. Failure reason: error while loading the trust store, error while reading certificates from "/home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt": open "/home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt": permission denied
WARN Signature sha256:11223344556677889900aabbccddeeff11223344556677889900aabbccddeeff failed verification with error: error while loading the trust store, error while reading certificates from "/home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt": open /home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt: permission denied
Error: signature verification failed for all the signatures associated with localhost:5001/net-monitor@sha256:000999888777666555444333222111fffeeedddcccbbbbaaa000999888777666
```

The following example shows there are no certificates in the `mystore` trust store.

```console
$ notation verify $IMAGE -v
...

> ERRO authenticity validation failed. Failure reason: error while loading the trust store, trust store "/home/yizha/.config/notation/truststore/x509/ca/mystore" has no x509 certificates

Reviewer 3: nit:
```suggestion
ERRO authenticity validation failed. Failure reason: error while loading the trust store, trust store "/home/USER/.config/notation/truststore/x509/ca/mystore" has no x509 certificates
```

Author: nice catch, fixed
